ng-dev/misc/update-yarn/cli.ts

```diff
@@ -0,0 +1,140 @@
+  const git = AuthenticatedGitClient.get();
+  /** The main branch name of the repository. */
+  const mainBranchName = git.mainBranchName;
+
+  const userFork = await git.getForkOfAuthenticatedUser();
+  if (userFork === null) {
+    error(
+      red(
+        'Unable to find a fork for the repository owned by you, please create a fork and try again',
+      ),
+    );
+    process.exitCode = 1;
+    return;
+  }
+  const localOwner = userFork.owner;
+
+  /** The original branch or ref before the command was invoked. */
+  const originalBranchOrRef = git.getCurrentBranchOrRevision();
+  if (git.hasUncommittedChanges()) {
+    error(
+      red(`Local working repository not clean. Please make sure there are no uncommitted changes.`),
+    );
+    process.exitCode = 1;
+    return;
+  }
+
+  const spinner = new Spinner('');
+  try {
+    spinner.update(`Fetching the latest primary branch from upstream: "${mainBranchName}"`);
+    git.run(['fetch', '-q', git.getRepoGitUrl(), mainBranchName]);
+    git.checkout('FETCH_HEAD', false);
+
+    /** Version numbers of the current (old) and new yarn installs. */
+    spinner.update(`Comparing current yarn version to the latest version on npm.`);
+    const yarnVersion = {
+      old: spawnSync('yarn', ['-v']).stdout.trim(),
+      new: spawnSync('yarn', ['info', 'yarn@latest', 'dist-tags.latest']).stdout.trim(),
+    };
+
+    if (yarnVersion.old === yarnVersion.new) {
+      spinner.complete();
+      error(red(`Yarn already updated to latest yarn (${yarnVersion.old})`));
+      process.exitCode = 1;
+      return;
+    }
+
+    /* File path of old yarn version */
+    const oldYarnPath = join(git.baseDir, `.yarn/releases/yarn-${yarnVersion.old}.cjs`);
+    const title = `build: update to yarn v${yarnVersion.new}`;
+    const body = `Update to the latest version of yarn, ${yarnVersion.new}.`;
+    const commitMessage = `${title}\n\n${body}`;
+    const {owner, repo} = git.remoteParams;
+    const prBranchName = `yarn-update-v${yarnVersion.new}`;
+
+    spinner.update(`Updating from yarn version '${yarnVersion.old}' to '${yarnVersion.new}'`);
+    spawnSync('yarn', ['policies', 'set-version', 'latest']);
+
+    spinner.update(`Removing previous yarn version (${yarnVersion.old})`);
+    unlinkSync(oldYarnPath);
+
+    spinner.update('Staging yarn vendoring files and creating commit');
+    git.run(['add', '.yarn/releases/**', '.yarnrc']);
+
+    /**
+     * New environment object with identifiers removed which prevent yarn from correctly loading
+     * new version from within a yarn call.
+     */
+    const env = {
+      ...process.env,
+      YARN_IGNORE_PATH: '0',
+      // The /tmp/yarn/ path needs to be removed when running the `git commit` command, because the
+      // environment in which this script runs, was executed by yarn and yarn binary forks itself.
+      // But since the yarn script it referenced was removed by this script, we need to run yarn
+      // "fresh".
```

Comment:
is that due to the husky hooks? e.g prepare-commit-msg? If so, I think we should just set HUSKY=0 similarly to how we do it in the merge and release tool

Reply:
Yes it is, and I created this change initially before we started using `HUSKY=0` and had forgotten about it.

Will updated to using that.